jerry-core/debugger/jerry-debugger.h

```diff
@@ -16,24 +16,72 @@
 #ifndef JERRY_DEBUGGER_H
 #define JERRY_DEBUGGER_H
 
-#define MAX_MESSAGE_SIZE 128
+#define MAX_BUFFER_SIZE 128
+
+char jerry_debugger_buffer[MAX_BUFFER_SIZE];
+
+/**
+ * Package header
+ */
+typdef struct
+{
+  jerry_debugger_header_type_t type;        /**< type of the message */
+  uint8_t size;                             /**< size of the message */
+} jerry_debugger_message_header_t;
 
 /**
-* Package header
-*/
+ * Types for the package
+ *
+ * It can help for the debugger who can decide what type
+ * of the data come from the JerryScript. If the buffer size it's not enough for
+ * the message the engine split up into many pieces. The last piece we know if the
+ * type has the '_END' postfix.
+ */
+typdef enum
+{
+  JERRY_DEBUGGER_SOURCE_FILE_NAME                /**< there is more piece of the source file name */
+  JERRY_DEBUGGER_SOURCE_FILE_NAME_END            /**< the last piece of the source file name */
+  JERRY_DEBUGGER_FUNCTION_NAME                   /**< there is more piece of the function name */
+  JERRY_DEBUGGER_FUNCTION_NAME_END               /**< the last piece of the function name */
+  JERRY_DEBUGGER_UNIQUE_START_BYTE_CODE_CPTR     /**< byte code starter compressed pointer */
+  JERRY_DEBUGGER_BREAKPOINT_LIST                 /**< there is more piece of the breakpoint list */
+  JERRY_DEBUGGER_BREAKPOINT_LIST_END             /**< the last piece of the breakpoint list */
+} jerry_debugger_header_type_t;
+
+/**
+ *  Source file name
+ */
 typdef struct
 {
-  uint8_t type;                     /**< type of the message */
-  uint8_t size;                     /**< size of the message */
-} jerry_debug_message_header_t;
+  jerry_debugger_message_header_t header;       /**< header of the source file name struct */
+  char file_name[1];                            /**< the message which contains the source file name */
```

Comment:
I would propose the same here: keep it shorter:
"source file name"
or
"JavaScript source file name"